File: src/main/java/com/example/counter/agent/AgentController.java
package com.example.counter.agent;

import org.springframework.web.bind.annotation.*;


@RestController
@RequestMapping("/api/agent")
public class AgentController {

    private static final org.slf4j.Logger logger = org.slf4j.LoggerFactory.getLogger(AgentController.class);

    private final AgentService agentService;

    public AgentController(AgentService agentService) {
        this.agentService = agentService;
    }

    @PostMapping("/ask")
    public AgentResponseDTO askAgent(
            @RequestBody AgentRequestDTO request,
            @RequestParam(name = "no-cache", required = false, defaultValue = "false") boolean noCache) {

        logger.info("Received incoming request: {}, no-cache: {}", request, noCache);

        return agentService.getResponse(request, noCache);
    }
}

File: src/main/kotlin/com/example/counter/agent/AgentController.kt
package com.example.counter.agent

import org.slf4j.LoggerFactory
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/api/agent")
class AgentController(private val agentService: AgentService) {

    private val logger = LoggerFactory.getLogger(AgentController::class.java)

    @PostMapping("/ask")
    fun askAgent(
        @RequestBody request: AgentRequestDTO,
        @RequestParam(name = "no-cache", required = false, defaultValue = "true") shouldUseCache: Boolean
    ): AgentResponseDTO {
        logger.info("Received incoming request: {}, shouldUseCache: {}", request, shouldUseCache)
        return agentService.getResponse(request, shouldUseCache)
    }
}
